static/js/inbox_ui.js

```diff
@@ -0,0 +1,732 @@
+import * as narrow from "./narrow";
+import * as narrow_state from "./narrow_state";
+import * as navigate from "./navigate";
+import * as people from "./people";
+import * as stream_data from "./stream_data";
+import * as stream_list from "./stream_list";
+import * as sub_store from "./sub_store";
+import * as top_left_corner from "./top_left_corner";
+import * as unread from "./unread";
+import * as unread_ui from "./unread_ui";
+import * as util from "./util";
+
+let pms_dict = {};
+let topics_dict = {};
+let streams_dict = {};
+let filtered_pm_list = {};
+let filtered_topic_list = {};
+let unread_pms_count;
+let unread_stream_msg_count;
+let filter_muted = false;
+let row_focus = 0;
+let stream_focus = 0;
+
+const ls_key = "inbox_mute_filter";
+const ls = localstorage();
+
+export let $current_focus_elem = "pm_header";
+let last_visited_topic = "";
+let last_visited_pm = "";
+
+function save_filter() {
+    ls.set(ls_key, filter_muted);
+}
+
+export function show() {
+    if (narrow.has_shown_message_list_view) {
+        narrow.save_pre_narrow_offset_for_reload();
+    }
+
+    if (is_visible()) {
+        return;
+    }
+
+    top_left_corner.narrow_to_inbox();
+    stream_list.handle_narrow_deactivated();
+
+    $("#message_feed_container").hide();
+    $("#inbox_view").show();
+    set_visible(true);
+    $("#message_view_header_underpadding").hide();
+    $(".header").css("padding-bottom", "0px");
+
+    unread_ui.hide_mark_as_read_turned_off_banner();
+    narrow_state.reset_current_filter();
+    narrow.set_narrow_title("Inbox");
+    message_view_header.render_title_area();
+    narrow.handle_middle_pane_transition();
+
+    complete_rerender();
+}
+
+export function hide() {
+    const $focused_element = $(document.activeElement);
+
+    if ($("#inbox_view").has($focused_element)) {
+        $focused_element.trigger("blur");
+    }
+
+    $("#message_view_header_underpadding").show();
+    $("#message_feed_container").show();
+
+    $("#inbox_view").hide();
+    set_visible(false);
+
+    $(".header").css("padding-bottom", "10px");
+
+    // This solves a bug with message_view_header
+    // being broken sometimes when we narrow
+    // to a filter and back to recent topics
+    // before it completely re-rerenders.
+    message_view_header.render_title_area();
+
+    // Fire our custom event
+    $("#message_feed_container").trigger("message_feed_shown");
+
+    // This makes sure user lands on the selected message
+    // and not always at the top of the narrow.
+    navigate.plan_scroll_to_selected();
+}
+
+function get_topic_key(stream_id, topic) {
+    return stream_id + ":" + topic;
+}
+
+function get_stream_key(stream_id) {
+    return "stream:" + stream_id;
+}
+
+function get_topic_row(topic_key) {
+    const $topic_row = $(`#${CSS.escape("inbox:" + topic_key)}`);
+    return $topic_row;
+}
+
+function get_topics_container(stream_id) {
+    const $topics_container = $(`#${CSS.escape("topics:" + stream_id)}`);
```

Comment:
I'm not sure about these IDs; they seem overly generic. I propose a pattern along the lines of `#topics:14` -> `#inbox_topics_for_stream_14`.

I this module is using these functions consistently, though, that should just require changes there.